ddl/column.go

```diff
@@ -1102,6 +1080,132 @@ func (w *worker) doModifyColumnTypeWithData(
 	return ver, errors.Trace(err)
 }
 
+func adjustTableInfoAfterModifyColumn(tblInfo *model.TableInfo, pos *ast.ColumnPosition,
+	oldCol, changingCol *model.ColumnInfo, newName model.CIStr, changingIdxs []*model.IndexInfo) (err error) {
+	internalColName := changingCol.Name
+	changingCol = replaceOldColumn(tblInfo, oldCol, changingCol, newName)
+	if len(changingIdxs) > 0 {
+		replaceOldIndexes(tblInfo, changingIdxs)
+		updateNewIndexesCols(tblInfo, internalColName, newName, changingCol.Offset)
+	}
```

Comment:
Do we need to do this check:https://github.com/pingcap/tidb/pull/33631/files#diff-42e0180cdc8c14def674a9e45d908d6700f6a8ccba0f9ff4b14b27c0daa49859R1602 here?